README.md

```diff
@@ -11,15 +11,282 @@ NATS Streaming provides the following high-level feature set.
 - Replay/Restart
 - Last Value Semantics
 
-### Getting Started
+# Table of Contents
+
+- [Important Changes](#important-changes)
+- [Concepts](#concepts)
+    * [Relation to NATS](#relation-to-nats)
+    * [Client Connections](#client-connections)
+    * [Channels](#channels)
+        * [Message Log](#message-log)
+        * [Subscriptions](#subscriptions)
+            * [Regular](#regular)
+            * [Durable](#durable)
+            * [Queue Group](#queue-group)
+            * [Redelivery](#redelivery)
+    * [Store Interface](#store-interface)
+    * [Clustering](#clustering)
+    * [Fault Tolerance](#fault-tolerance)
+        * [Active Server](#active-server)
+        * [Standby Servers](#standby-servers)
+        * [Shared State](#shared-state)
+        * [Failover](#failover)
+- [Getting Started](#getting-started)
+    * [Building](#building)
+    * [Running](#running)
+- [Configuring](#configuring)
+    * [Command line arguments](#command-line-arguments)
+    * [Configuration file](#configuration-file)
+    * [Store Limits](#store-limits)
+        * [Limits inheritance](#limits-inheritance)
+    * [Securing](#securing)
+        * [Authorization](#authorization)
+        * [TLS](#tls)
+    * [Persistence](#persistence)
+        * [File Store](#file-store)
+            * [File Store Options](#file-store-options)
+- [Clients](#clients)
+- [Licence](#license)
+
+# Important Changes
+
+## Version `0.4.0`
```

Comment:
This is not a change log per-se. Most of the time there will be no addition to that section. This is just when we either break backward compatibility or introduce/change Store interface for people that have done (or planning to do) store implementations.